packages/core/src/types/future.ts

```diff
@@ -70,6 +78,7 @@ export interface EventParamFuture {
   vertexId: number;
   label: string;
   type: "eventParam";
+  subtype: string;
```

Comment:
Can you walk me through the different values this can be set to?

Reply:
Any solidity type except `mapping`, I believe. So, `uint8..256`, `int8..256`, `fixed8..256`, `address`, `bool`, `bytes`, and `bytes1..32`, also array versions of all of those as well as `tuple` which is how `struct`s are represented as well.

Reply:
I think we eventually could use these to make validation for `awaitEvent` much more robust.

Reply:
Ah I see. So we use the abi to derive the subtype and then we can use that in validation - the abi says this is a uint256 but your trying to passin it as an arg to a function that takes strings?